### ISSUE TITLE
sct_extract_metric.py --> needs to be clarified

### ISSUE THREAD

**Issue by tanguyduval** [neuropoly/spinalcordtoolbox#736]
Couple of issues:

- [ ] `sct_extract_metric.py -f $SCT_DIR/data/template` is a wrong example.. T2-weighted template is part of it and is not a partial volume mask.. --> AGREED
- [ ] `-l wm` will work only if we use `-f $SCT_DIR/data/atlas` because it assumes CSF label is 31 --> AGREED. WE SHOULD ONLY USE THE FOLDER ATLAS.
- [ ] `-l` option is unclear.. Are the other tracts in the list averaged to compute the MAP?? --> YES, BUT MAP IS ONLY CORRECT WHEN USING ATLAS FOLDER
- [ ] `-a` Does averaged before or after MAP? --> NO, IT CONCATENATES THE TRACTS AND THEN RUNS MAP (BETTER)
- [x] output file metric_label.txt has a STD that is always 0 when using MAP method. Conterintuitive. Why not computing the std in the tract. --> BECAUSE ML OR MAP ESTIMATES THE "TRUE" VALUE (BETA IS THE ESTIMATE OF THE EQUATION: beta=(xt.x)-1.xt.y, SO THERE IS NO  #STD).
- [ ] Propose a .mat file for output. Or in the doc refer to `spinalcordtoolbox/matlab/register2template_SliceBySlice/sct_extract_metric_read.m`

**Comment by tanguyduval** [comment]
I propose to use `-l` option for selecting the tracts that will be averaged for the MAP... and remove `-a` 
Examples: 
`-l 0:29` will extract the WM
`-l [0,1,15,16],[2:6],[17:21],[7:14,22:19]` will extract the dorsal, left column, right column and ventral column.

**Comment by tanguyduval** [comment]
Or maybe better. In the info_label.txt, add a section "Combined tracts":
`````
# White matter atlas - generated on 2014-12-09
# ID, name, file
0, left fasciculus gracilis, WMtract__00.nii.gz
1, left fasciculus cuneatus, WMtract__01.nii.gz
2, left lateral corticospinal tract, WMtract__02.nii.gz
3, left spinocerebellar tract, WMtract__03.nii.gz
4, left rubrospinal tract, WMtract__04.nii.gz
5, left lateral reticulospinal tract, WMtract__05.nii.gz
6, left spinal lemniscus (spinothalamic and spinoreticular tracts), WMtract__06.nii.gz
7, left spino-olivary tract, WMtract__07.nii.gz
8, left ventrolateral reticulospinal tract, WMtract__08.nii.gz
9, left lateral vestibulospinal tract, WMtract__09.nii.gz
10, left ventral reticulospinal tract, WMtract__10.nii.gz
11, left ventral corticospinal tract, WMtract__11.nii.gz
12, left tectospinal tract, WMtract__12.nii.gz
13, left medial reticulospinal tract, WMtract__13.nii.gz
14, left medial longitudinal fasciculus, WMtract__14.nii.gz
15, right fasciculus gracilis, WMtract__15.nii.gz
16, right fasciculus cuneatus, WMtract__16.nii.gz
17, right lateral corticospinal tract, WMtract__17.nii.gz
18, right spinocerebellar tract, WMtract__18.nii.gz
19, right rubrospinal tract, WMtract__19.nii.gz
20, right lateral reticulospinal tract, WMtract__20.nii.gz
21, right spinal lemniscus (spinothalamic and spinoreticular tracts), WMtract__21.nii.gz
22, right spino-olivary tract, WMtract__22.nii.gz
23, right ventrolateral reticulospinal tract, WMtract__23.nii.gz
24, right lateral vestibulospinal tract, WMtract__24.nii.gz
25, right ventral reticulospinal tract, WMtract__25.nii.gz
26, right ventral corticospinal tract, WMtract__26.nii.gz
27, right tectospinal tract, WMtract__27.nii.gz
28, right medial reticulospinal tract, WMtract__28.nii.gz
29, right medial longitudinal fasciculus, WMtract__29.nii.gz
30, gray matter, WMtract__30.nii.gz
31, CSF contour, WMtract__31.nii.gz

# Combined tracts
# ID, name, IDs
32, White Matter, 0:29
33, Dorsal, 0,1,15,16
34, Lateral Left, 2:6
35, Lateral Right, 17:21
36, Ventral, 7:14,22:19

````

**Comment by tanguyduval** [comment]
@jcohenadad Peut-on changer ça avec cette dernière solution? Les flags `-l` et `-a` disparaitront de la doc mais seront compatible avec les anciennes version. Le script sera plus clair

**Comment by jcohenadad** [comment]
yes, this is a good idea. But we should leave the possibility to do, e.g. 2,3,4 (currently not listed in the combined tracts). So maybe we can keep the "-l" with [0,1,15,16],[2:6]

**Comment by tanguyduval** [comment]
If the user wants to do 2,3,4 he can add at the end of info_label.txt.
````
32, Motor Tracts, 2,3,4
````

Reason for this: 
1- Output needs a label (e.g. Motor Tract) to be consistent, so that it can be read easily:
````
...
30, gray matter, 0.219579, 0.000000
31, CSF contour, 0.072080, 0.000000
32, Motor Tract, 0.072080, 0.000000
...
````

2- redundancy of options... (apple/google spirit :-))

**Comment by jcohenadad** [comment]
ok it makes sense. Let's discuss it in the next SCT meeting

**Comment by jcohenadad** [comment]
todo: 
[ ] remove flags -l and -a and output result for all labels
[ ] remove info_label in /template and check if user is under /atlas --> if not, exit.
[ ] add flag -output-type {mat, xls, txt} --> default txt, separate with ","

**Comment by slevyrosetti** [comment]
Issue was addressed on branch sl_sct_extract_metric in commit 435c1cd.
The implementation was completed in sct_extract_metric but since other scripts use functions from sct_extract_metric that have been modified (read_label_file, save_metric), these scripts (sct_warp_template, sct_process_segmentation) have to be updated accordingly.

N.B.: Concerning point #4, the output file type .mat (for matlab use) has not been allowed on purposes because it would be redundant with the .xls type since plenty of matlab functions enables to load .xls files directly and easily.

**Comment by jcohenadad** [comment]
thank you Simon, could you please update sct_warp_template, sct_process_segmentation in your branch?

**Comment by slevyrosetti** [comment]
Yes I'm on it.

**Comment by slevyrosetti** [comment]
The second check box is a bit problematic because of the script sct_warp_template which reads the file info_label.txt to get the names of the files to warp. For now, I will remove the check `user under /atlas --if not, exit`

**Comment by slevyrosetti** [comment]
@jcohenadad : could you add the file info_label.txt attached to the folder $SCT_DIR/data/atlas please? It integrates the combined labels.
[info_label.txt](https://github.com/neuropoly/spinalcordtoolbox/files/185051/info_label.txt)

**Comment by slevyrosetti** [comment]
All scripts using functions of sct_extract_metric that were modified are now up-to-date on branch sl_sct_extract_metric.

**Comment by slevyrosetti** [comment]
Merged via commit 7f6d9fb.
